target-app/Home/ViewModels/HomeViewModel.swift

```diff
@@ -10,9 +10,27 @@ import CoreLocation
 
 class HomeViewModel {
     
+    // MARK: - Observed Properties
+    @Published private var state: AuthViewModelState = .network(state: .idle)
     @Published var locationTapped: CLLocation?
+    @Published var targets: [TargetElement]?
     
     func setLocationTapped(withLocation location: CLLocation) {
         self.locationTapped = location
     }
+    
+    func getTargets() {
+        state = .network(state: .loading)
+        TargetServices.get() { [weak self] result in
+            guard let self = self else { return }
+            
+            switch result {
+            case .success(let targets):
+                self.state = .loggedIn
```

Comment:
Is this view model supposed to handle authentication states? AFAIK the map view is only available for logged in users.
It is confusing also to set the `loggedIn` state when you fetch targets.

Reply:
Thats true. Removed it!